codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -545,6 +545,7 @@ void takeChild(Iterator<IntObjectMap.PrimitiveEntry<State>> childItr, State chil
          * Removes the child priority and moves any of its dependencies to being direct dependencies on this node.
          */
         void removeChild(State child) {
+            long totalWeight = child.getTotalWeight();
```

Comment:
Could calculate the total weight only when necessary. How about moving this right before the `while` block below? We could also skip the entire loop and calculation if `child.children` is empty.

Reply:
yeah, and thanks @trustin  for your advices, I'll make a change about it